### DISCUSSION TITLE
[Feature] (Automatic) video captions/subtitles

### DISCUSSION THREAD

**Discussion by muellermartin** [immich-app/immich#14658] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

For videos it would be very useful to have captions and/or subtitles which are also searchable.

Usually videos do not have captions/subtitles, but with tools like [OpenAI Whisper](https://github.com/openai/whisper) and [whisper.cpp](https://github.com/ggerganov/whisper.cpp) videos can be automatically transcribed.

This would make videos searchable via full text search to find contents and even jump to the part in the video.

The captions can be displayed via the [\<track\>](https://developer.mozilla.org/en-US/docs/Web/Media/Audio_and_video_delivery/Adding_captions_and_subtitles_to_HTML5_video) HTML element and WebVTT files (e.g. which can be obtained via whisper).

### Platform

- [X] Server
- [X] Web
- [X] Mobile

**Comment by MaKraMc** [comment]
I like this Idea. This would make the search even more powerful!

**Comment by BadQuantaGames** [comment]
+1 for this, would be extremely useful for advance searching.

**Comment by cove** [comment]
Even if it doesn't do the encoding, it would still be helpful if it could honor the subtitles side cars. e.g. .srt .ass .vtt

**Comment by Mexallon** [comment]
I would love the feature of subtitles in Immich. I got plenty of old videos from my dad filming and put a lot of work to generate subtitles to display the current date and time. There are often time jumps in the video and this makes tracking time easier.

**Comment by bo0tzz** [comment]
#13503

**Comment by 2q2code** [comment]
Definitely want this feature. That's the way my memory works when searching for a video. It's just as often about words said in it as it is the visual content. I've written something as a POC in the past for indexing movies by their dialogue. It's funny to count the f-bombs in films. 😁
